### PR TITLE
Extract autoloading process

### PR DESCRIPTION
**Pull request by aledeg**
Changes proposed in this pull request:

- Change autoloading to accomodate namespaces

How to test the feature manually:

1. Use the application to check than everything is working as before.

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/master/docs/en/developers/04_Pull_requests.md).

### CODE REVIEW

File: lib/lib_rss.php [hunk 1]
```diff
@@ -15,6 +15,8 @@ function mb_strcut($str, $start, $length = null, $encoding = 'UTF-8') {
 	openlog('FreshRSS', LOG_CONS | LOG_ODELAY | LOG_PID, LOG_USER);
 }
 
+require_once 'autoload.php';
```
Comment (Alkarex) [inline lib/lib_rss.php]:
```suggestion
require_once LIB_PATH . '/autoload.php';
```

Just a suggestion to be more similar to what we do elswhere

File: lib/autoload.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * Register path in which to look for a class.
+     */
+    public function registerPath($path) {
+        if (is_string($path)) {
+            $this->searchPath[] = $path;
+        }
+        if (is_array($path)) {
+            array_push($this->searchPath, ...$path);
```
Comment (Alkarex) [inline lib/autoload.php:14]: I have just double-checked that the splat operator is available from PHP 5.6+, so fine

**Review by Alkarex — APPROVED** [review]

File: lib/autoload.php (hunk 2 above)
Comment (aledeg) [reply lib/autoload.php:14]: I've checked that before writing it :)

File: lib/lib_rss.php (hunk 1 above)
Comment (aledeg) [reply lib/lib_rss.php]: Fixed

**Pull request merged**